deployment_report/model/utils/config_util.py

```diff
@@ -54,66 +73,175 @@ def get_cadence_version(config_maps: Dict) -> Optional[Text]:
+            db_dict = _get_db_info_v1(config_maps=config_maps)
+
+        except KeyError:
+            db_dict[_UNSUPPORTED_DB_] = {_TYPE_KEY_: _UNSUPPORTED_DB_}
+
+    return db_dict
+
+
+def _get_db_info_v1(config_maps: Dict) -> Dict:
+    """
+    Returns the db information of the targeted SAS deployment.
+
+    :param config_maps: The ConfigMap resources to evaluate. This is before data server operator change.
 
     :return: A dict representing the db information of the targeted SAS deployment, or None if the given
              ConfigMap does not contain the requested information.
     """
-    # look for the necessary ConfigMap
+    # initialize the return value
+    db_dict: Optional[Dict] = dict()
+
     for config_map_name, config_map_details in config_maps.items():
+        dbs: Optional[Dict] = dict()
+
+        # look for the necessary ConfigMap
         if _SAS_POSTGRES_CONFIG_CONFIG_MAP_NAME_ in config_map_name:
             # get the KubernetesResource
             config_map: KubernetesResource = \
-                config_map_details[ViyaDeploymentReportKeys.ResourceDetails.RESOURCE_DEFINITION]
+                config_map_details[Keys.ResourceDetails.RESOURCE_DEFINITION]
 
             # get the ConfigMap data
+            db_name: Optional[Text] = config_map.get_name()
             db_data: Optional[Dict] = config_map.get_data()
+            if db_data:
+                try:
+                    # check whether the db configuration is external
+                    if db_data[_EXTERNAL_DATABASE_KEY_] == "false":
+                        dbs = {_TYPE_KEY_: _INTERNAL_DB_}
+                    else:
+                        # if external, create the dict of all relevant info
+                        dbs = {
+                            _TYPE_KEY_: _EXTERNAL_DB_,
+                            _HOST_KEY_: db_data[_DATABASE_HOST_KEY_],
+                            _PORT_KEY_: db_data[_DATABASE_PORT_KEY_],
+                            _NAME_KEY_: db_data[_DATABASE_NAME_KEY_],
+                            _SSLENABLED_KEY_: db_data[_DATABASE_SSL_ENABLED_KEY_]
+                        }
+                except KeyError:
+                    try:
+                        if _CRUNCH_DB_ in db_data[_ALT_DB_]:
+                            dbs = {_TYPE_KEY_: _INTERNAL_DB_}
+                    except KeyError:
+                        # if an expected key wasn't defined, continue
+                        continue
+        if dbs:
+            # convert db_name to be aligned with v2
+            if _DBNAME_CONFIG_POSTGRES_ in db_name:
+                db_name = _DBNAME_POSTGRES_
+            elif _DBNAME_CONFIG_CPSPOSTGRES_ in db_name:
+                db_name = _DBNAME_CPSPOSTGRES_
+
+            db_dict[db_name] = dbs
+
+    return db_dict
+
+
+def _get_db_info_v2(pgclusters: Dict) -> Dict:
+    """
+    Returns the db information of the targeted SAS deployment.
+
+    :param pgclusters: The pgclusters resource to evaluate. (RDCHANGE-434)
```

Comment:
This might be a reference to an internal Jira ticket. We should remove references to internal tickets from public repos.